packages/sdk/src/across/clients/bridgePool.ts

```diff
@@ -297,10 +297,6 @@ export class ReadPoolClient {
 export function validateWithdraw(pool: Pool, user: User, lpTokenAmount: BigNumberish) {
   const l1TokensToReturn = BigNumber.from(lpTokenAmount).mul(pool.exchangeRateCurrent).div(fixedPointAdjustment);
   assert(BigNumber.from(l1TokensToReturn).gt("0"), "Must withdraw amount greater than 0");
-  assert(
```

Comment:
so there is a situation where this does not error and a user cant withdraw, no? if you want to use the contracts to validate this could you not try `.call`ing `removeLiquidity` with the `from` set to the users wallet and see if it reverts?

Reply:
it would be great if i could do that check without calling the contract, is there a way given the known vars?

Reply:
_Technically_ you could probably do this. But operationally, this would be very difficult. The contract does some math based on its balance and the current liquid reserves, bond, and utilized reserves to figure out what the new utilized reserves will be.

Generally, the easiest, most accurate, and most future-proof way to do this is to run a .call() (I know this is inconvenient for the FE since it's async and depends on what the user types in).

Reply:
ok interesting. so i could change this validation to just do all the validation in contract in that case, since im sure the checks in the code here will be redundant

Reply:
I tested out using the contract to validate, but what i dont like is that i dont get back good error messages. I can try to clean up the error when it throws, but i would basically get this `ERC20: burn amount exceeds balance` when withdrawing more eth than deposited. Im not sure how to test for withdrawing more liquidity than what is available. For now i will add improving this function to the backlog and keep this client side implementation for now.

Reply:
https://app.shortcut.com/uma-project/story/3637/improve-pool-withdraw-validation

Reply:
SGTM!